torch/quantization/observer.py

```diff
@@ -115,7 +115,11 @@ def _calculate_per_channel_qparams(self, min_vals, max_vals):
             )
             scales[i] = float(qparam[0])
             zero_points[i] = int(qparam[1])
-
+        # Convert scale to float precision
+        # Needed to ensure that fake quantization module and operator level numerics match
+        # for test:
+        # caffe2/test:fake_quant - test_fq_module (test_fake_quant.TestFakeQuantizePerChannel)
+        scales = scales.to(torch.float)
```

Comment:
you already initialize `scales` with torch.ones() above which is float. Why is it needed?

also remove the reference to the test - it's not helpful